### PR TITLE
refactor: add list dev tunnels

### PR DESCRIPTION
**Pull request by xiaolang124**
1. Add list dev tunnels API for VS.

### CODE REVIEW

File: packages/fx-core/package.json [hunk 1]
```diff
@@ -121,7 +121,10 @@
     "validator": "^13.7.0",
     "xml2js": "^0.5.0",
     "yaml": "^2.2.2",
-    "yaml-language-server": "1.7.0"
+    "yaml-language-server": "1.7.0",
+    "@microsoft/dev-tunnels-connections": "~1.0.7360",
```
Comment (swatDong) [inline packages/fx-core/package.json:125]: Is `@microsoft/dev-tunnels-connections` unused that can be removed?

File: packages/fx-core/package.json [hunk 2]
```diff
@@ -121,7 +121,10 @@
...
+    "@microsoft/dev-tunnels-contracts": "~1.0.7360",
+    "@microsoft/dev-tunnels-management": "~1.0.7360"
```
Comment (swatDong) [inline packages/fx-core/package.json:127]: Update *package-lock.json* file as well?

**Review by swatDong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/package.json (hunk 1 above)
Comment (xiaolang124) [reply packages/fx-core/package.json]: Done.

File: packages/fx-core/package.json (hunk 2 above)
Comment (xiaolang124) [reply packages/fx-core/package.json:126]: Done.

**Pull request merged**